### ISSUE TITLE
The auto-indentation of vhdl filetype works weirdly in Vim

### ISSUE THREAD

**Issue by wtswkz** [vim/vim#9265]
### Steps to reproduce

Myself encountered the problem with the **if** statement, and it's also found by others that there are also some problems with entity declaration.

1. The problem with the **if** statement is as follows:

> When I type "if xxx then" and Enter, it looks like as follows:
![image](https://user-images.githubusercontent.com/6916435/144613311-8231f935-728c-4443-823e-7a949a236d8c.png)

> The I type "end" and Space in a new line, then the "end" auto aligned with the "if", it looks like:
![image](https://user-images.githubusercontent.com/6916435/144613547-da927eb3-819c-4a05-9a04-057f90de81c4.png)

> When I finish the statement with "if;", the auto-alignment of "end" is canceled, it looks like:
![image](https://user-images.githubusercontent.com/6916435/144613783-d5430198-38e0-4816-808e-f14170b359ec.png)

This problem won't occur if I type the **if** statement line by line, which means insert a non-empty line first, such as an assignment statement or anything else that I want to execute if the condition is true, before closing the **if** statement with **end if;**. It's my habit to close the **if** statement as soon as I have typed `if xxx then`, then add other statements. Similar issue also exists with the empty **entity** declaration, the closing "end ENTITY_NAME;" won't auto-indent there is no "port" declaration or "generic" declaration.

2. The problem with the **port** declaration is as follows:

> I have declared a port as follows:
![image](https://user-images.githubusercontent.com/6916435/144616283-f77b0a56-0a8d-4250-b810-b6c24fcb6e89.png)
> Then I hit Enter to declare another port, the cursor is aligned with the "port", but not with the port name in the previous line:
![image](https://user-images.githubusercontent.com/6916435/144616570-46d6d8c3-dbf2-4b4f-ba9a-fa08a5906f1a.png)

Well, These are the indentation problems with filetype of vhdl that I found by now.

### Expected behaviour

I know the problems that I post in this issue have something to do with my habit of coding. But I still post this to seek a better vhdl indentation feature that auto-indentation works fine no mater what the coding habit of people is.

### Operating system

Fedora 35, KDE

### Version of Vim

VIM 8.2, konsole 21.08.3

**Comment by chrisbra** [comment]
Have you tried contacting the indent script maintainer mentioned at the top of the file?

https://github.com/vim/vim/blob/04b7b4bf7fe57375bf3c4ed3fd087fc0bb0147ba/runtime/indent/vhdl.vim#L1-L6

**Comment by wtswkz** [comment]
No. I didn’t think of that. Thanks for reminding me! I’ll send the author an email.
